src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NFloat.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+
+#pragma warning disable SA1121 // We use our own aliases since they differ per platform
+#if TARGET_32BIT
+using NativeType = System.Single;
+#else
+using NativeType = System.Double;
+#endif
+
+namespace System.Runtime.InteropServices
+{
+    [Intrinsic]
+    public readonly struct NFloat : IEquatable<NFloat>
```

Comment:
Mono's `nfloat` implements more interfaces:
https://github.com/dotnet/runtime/blob/dada1303ec3367e194144974d049eead8c9d11d1/src/mono/mono/mini/builtin-types.cs#L2478
same as:
https://github.com/dotnet/runtime/blob/57ba63838066642784339bcbee23992c737e44f9/src/libraries/System.Private.CoreLib/src/System/Single.cs#L26
would it make sense for new types to implement those as well?

Reply:
These types are meant for interop only. It is intentionally that they do not implement all interfaces implemented by the regular primitive types.

Reply:
Right, these are meant purely for simplifying internal interop definitions and not for more general public consumption.

It's always possible to revisit if we find this is hindering in real world scenarios, but it's at least initially not thought to be that way.